packages/launchdarkly-adapter/modules/adapter/adapter.ts

```diff
@@ -77,8 +80,15 @@ const setupFlagSubcription = ({
           [normalizedFlagName]: normalizedFlagValue,
         };
 
-        updateFlagsInAdapterState(flags);
-        onFlagsStateChange(flags);
+        const updateFlags = () => {
+          updateFlagsInAdapterState(flags);
+          onFlagsStateChange(flags);
+        };
+
+        debounce(updateFlags, {
+          wait: flagsUpdateDelayMs,
+          immediate: !flagsUpdateDelayMs,
+        })();
```

Comment:
This returns a `fn` which you can `.cancel` again but we don't need that. The library's `wait` and `immediate` options allow us to avoid to `if-else` on the argument.